lib/tool_shed/webapp/api/repositories.py

```diff
@@ -552,10 +552,17 @@ def reset_metadata_on_repositories(self, trans, payload, **kwd):
 
         """
 
-        def handle_repository(trans, rmm, repository, results):
+        def handle_repository(trans, repository, results):
             log.debug(f"Resetting metadata on repository {repository.name}")
             try:
-                rmm.set_repository(repository)
+                rmm = repository_metadata_manager.RepositoryMetadataManager(
```

Comment:
I've looked at it couple time now, and I haven't figure out why exactly this fixes the bug. Is it because there's still something that `set_repository` didn't correctly reset to the right repository ? Given all the things the init method does this is definitely better, just curious at this point.

Reply:
😆 It isn't obvious at all - it took hours of debugging. set_repository resets resetting_all_metadata_on_repository to ``False``, which has... implications for how working directories are used in ``generate_metadata_for_changeset_revision``. When it resets, the actual directory of the repository is used and only the most recent contents of the repository are used as the working directory to take metadata from. So the tool shed is like "heres is all my tools and versions" over and over for each revision but the contents used to generate that are only the latest revision... and so only one interesting metadata repository is discovered.